hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/TransferContextImpl.java

```diff
@@ -93,7 +96,21 @@ public void createFromAlias(final Bytes alias, final int reqMaxAutoAssociations)
             validateTrue(tokensConfig.autoCreationsIsEnabled(), NOT_SUPPORTED);
         }
         // Keep the created account in the resolutions map
-        final var createdAccount = autoAccountCreator.create(alias, reqMaxAutoAssociations);
+        AccountID createdAccount;
+        try {
+            createdAccount = autoAccountCreator.create(alias, reqMaxAutoAssociations);
+        } catch (HandleException e) {
+            if (getHandleContext().getHederaFunctionality() == CRYPTO_TRANSFER
```

Comment:
⚠️ Hmm...the `HandleContext` given to the `CryptoTransferHandler` is _always_ going to have a functionality of `CRYPTO_TRANSFER`.

And its category will never be `PRECEDING` (only synthetic `CRYPTO_CREATE` records are ever preceding).

_However_, your implementation of `isSelfSubmitted()` will only be true in the exact conditions when we want to reclaim throttle capacity, since synthetic `TransactionBody`'s never have `nodeAccountID` set.

So we can use just,
```
if (getHandleContext().isSelfSubmitted()) {
    final int autoCreationsNumber = numOfAutoCreations() + numOfLazyCreations();
    getHandleContext().reclaimPreviouslyReservedThrottle(autoCreationsNumber, CRYPTO_CREATE);
}
```
and that should be what we want! 🙌

Reply:
Ahh got it!
Thanks for clarifying all of that Michael 🙌